packages/cosmic-swingset/src/helpers/read-fd-lines.js

```diff
@@ -0,0 +1,17 @@
+// @ts-check
+import fs from 'fs';
+import readline from 'readline';
+
+const debug = (..._args) => {}; // console.error(..._args);
+
+export const makeReadlineLoop = async (fd, handler) => {
+  const rl = readline.createInterface({
```

Comment:
`readline` has horrendous performance, which is why I adapted `BufferLineTransform`.